From 4c0177328e8ec1223fea45c8f37000af4eb2178b Mon Sep 17 00:00:00 2001
From: Tim Waugh <twaugh@redhat.com>
Date: Tue, 9 Aug 2016 18:11:46 +0200
Subject: Set the default for SyncOnClose to Yes

Last-Update: 2015-02-10

Patch-Name: cupsd-set-default-for-SyncOnClose-to-Yes.patch
---
 conf/cups-files.conf.in           | 2 +-
 doc/help/man-cups-files.conf.html | 2 +-
 man/cups-files.conf.man.in        | 2 +-
 scheduler/conf.c                  | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/conf/cups-files.conf.in b/conf/cups-files.conf.in
index 4a78ba615..a3d088da5 100644
--- a/conf/cups-files.conf.in
+++ b/conf/cups-files.conf.in
@@ -7,7 +7,7 @@
 #FatalErrors @CUPS_FATAL_ERRORS@
 
 # Do we call fsync() after writing configuration or status files?
-#SyncOnClose No
+#SyncOnClose Yes
 
 # Default user and group for filters/backends/helper programs; this cannot be
 # any user or group that resolves to ID 0 for security reasons...
diff --git a/doc/help/man-cups-files.conf.html b/doc/help/man-cups-files.conf.html
index 42dae48b3..92812db5a 100644
--- a/doc/help/man-cups-files.conf.html
+++ b/doc/help/man-cups-files.conf.html
@@ -148,7 +148,7 @@ The default is "/var/run/cups" or "/etc/cups" depending on the platform.
 <dd style="margin-left: 5.0em">Specifies whether the scheduler calls
 <b>fsync</b>(2)
 after writing configuration or state files.
-The default is "No".
+The default is "Yes".
 <dt><a name="SystemGroup"></a><b>SystemGroup </b><i>group-name </i>[ ... <i>group-name</i> ]
 <dd style="margin-left: 5.0em">Specifies the group(s) to use for <i>@SYSTEM</i> group authentication.
 The default contains "admin", "lpadmin", "root", "sys", and/or "system".
diff --git a/man/cups-files.conf.man.in b/man/cups-files.conf.man.in
index 2ed468661..07446abd6 100644
--- a/man/cups-files.conf.man.in
+++ b/man/cups-files.conf.man.in
@@ -204,7 +204,7 @@ The default is "/var/run/cups" or "/etc/cups" depending on the platform.
 Specifies whether the scheduler calls
 .BR fsync (2)
 after writing configuration or state files.
-The default is "No".
+The default is "Yes".
 .\"#SystemGroup
 .TP 5
 \fBSystemGroup \fIgroup-name \fR[ ... \fIgroup-name\fR ]
diff --git a/scheduler/conf.c b/scheduler/conf.c
index ae65d134f..bd1bac436 100644
--- a/scheduler/conf.c
+++ b/scheduler/conf.c
@@ -735,7 +735,7 @@ cupsdReadConfiguration(void)
   RootCertDuration         = 300;
   Sandboxing               = CUPSD_SANDBOXING_STRICT;
   StrictConformance        = FALSE;
-  SyncOnClose              = FALSE;
+  SyncOnClose              = TRUE;
   Timeout                  = DEFAULT_TIMEOUT;
   WebInterface             = CUPS_DEFAULT_WEBIF;
 
